File: Server/route/ctfChallenge.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetCTFChallengeRoute(r *gin.Engine) {
	adminCTFChallenge := r.Group("/ctfChallenge")
	adminCTFChallenge.Use(middleware.AdminAuthMiddleware())
	{
		adminCTFChallenge.POST("/GetCTFChallengeList", api.GetCTFChallengeList)
		adminCTFChallenge.POST("/GetCTFChallenge", api.GetCTFChallenge)
		adminCTFChallenge.POST("/CreateCTFChallenge", api.CreateCTFChallenge)
		adminCTFChallenge.POST("/UpdateCTFChallenge", api.UpdateCTFChallenge)
		adminCTFChallenge.POST("/DeleteCTFChallenge", api.DeleteCTFChallenge)
	}
}

File: Web/src/api/ctfChallenge.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetCTFChallengeList(data) {
  return apiClient.post("/ctfChallenge/GetCTFChallengeList", data).then(handleResponse).catch(handleError)
}

function GetCTFChallenge(data) {
  return apiClient.post("/ctfChallenge/GetCTFChallenge", data).then(handleResponse).catch(handleError)
}

function CreateCTFChallenge(data) {
  return apiClient.post("/ctfChallenge/CreateCTFChallenge", data).then(handleResponse).catch(handleError)
}

function UpdateCTFChallenge(data) {
  return apiClient.post("/ctfChallenge/UpdateCTFChallenge", data).then(handleResponse).catch(handleError)
}

function DeleteCTFChallenge(data) {
  return apiClient.post("/ctfChallenge/DeleteCTFChallenge", data).then(handleResponse).catch(handleError)
}

const ctfChallengeApi = {
  GetCTFChallengeList,
  GetCTFChallenge,
  CreateCTFChallenge,
  UpdateCTFChallenge,
  DeleteCTFChallenge
}

export default ctfChallengeApi
